src/core/model.jl

```diff
@@ -108,56 +106,57 @@ function AgentBasedModel(
     agent_validator(A, space, warn)
 
     agents = Dict{Int, A}()
-    return ABM{A, S, F, P}(agents, space, scheduler, properties)
+    return ABM{A, S, F, P}(agents, space, scheduler, properties, [0])
 end
 
 function AgentBasedModel(agent::AbstractAgent, args...; kwargs...)
     return ABM(typeof(agent), args...; kwargs...)
 end
 
-function Base.show(io::IO, abm::ABM{A}) where {A}
-    n = isconcretetype(A) ? nameof(A) : string(A)
-    s = "AgentBasedModel with $(nagents(abm)) agents of type $(n)"
-    if abm.space == nothing
-        s*= "\n no space"
-    else
-        s*= "\n space: $(sprint(show, abm.space))"
-    end
-    s*= "\n scheduler: $(schedulername(abm.scheduler))"
-    print(io, s)
-    if abm.properties ≠ nothing
-        print(io, "\n properties: ", abm.properties)
-    end
-end
-schedulername(x::Union{Function, DataType}) = nameof(x)
-schedulername(x) = string(x)
-
+#######################################################################################
+# %% Model accessing api
+#######################################################################################
+export random_agent, nagents, allagents, allids
 
 """
     getindex(model::ABM, id::Integer)
+    model[id]
 
 Return an agent given its ID.
 """
 Base.getindex(m::ABM, id::Integer) = m.agents[id]
 
 """
     setindex!(model::ABM, agent::AbstractAgent, id::Int)
+    model[id] = agent
 
-Add an `agent` to the `model` at a given index: `id`. Note this method will return an error if the `id` requested is not equal to `agent.id`.
+Add an `agent` to the `model` at a given index: `id`.
+Note this method will return an error if the `id` requested is not equal to `agent.id`.
+**Internal method**, ose [`add_agents!`](@ref) instead to actually add an agent.
 """
 function Base.setindex!(m::ABM, a::AbstractAgent, id::Int)
     a.id ≠ id && throw(ArgumentError("You are adding an agent to an ID not equal with the agent's ID!"))
     m.agents[id] = a
+    m.maxid[1] < id && (m.maxid[1] += 1)
+    return a
 end
 
+"""
+    nextid(model::ABM) → id
+Return a valid `id` for creating a new agent with it.
+"""
+nextid(model::ABM) = model.maxid[1] + 1
+
 """
     getproperty(model::ABM, prop::Symbol)
+    model.prop
 
-Return a property from the current `model`. Possibilities are
+Return a property from the current `model`, assuming the model properties are i  Possibilities are
```

Comment:
Think something got cut off in this docstring addition.

Reply:
fixed!